src/Features/LanguageServer/Protocol/LanguageServerTarget.cs

```diff
@@ -131,6 +142,10 @@ public Task<InitializeResult> InitializeAsync(InitializeParams initializeParams,
 
                 Contract.ThrowIfTrue(_clientCapabilities != null, $"{nameof(InitializeAsync)} called multiple times");
                 _clientCapabilities = initializeParams.Capabilities;
+
+                // TO-DO: Add client capability check below once LSP side is merged
+                _semanticTokensRefreshListener = new SemanticTokensRefreshListener(_lspWorkspaceManager, _jsonRpc, _listener, cancellationToken);
```

Comment:
I recommend hooking this up in `Initialized` so that we cannot send notifications before we have sent back the initialize result